backend/apps/guests/services.py

```
@@ -196,6 +196,38 @@ def create_guest_pass_for_resident(
     return guest_pass
 
 
+def cancel_guest_pass_for_resident(pass_id: int, membership: Membership, residence) -> GuestPass:
+    try:
+        guest_pass = GuestPass.objects.get(
+            id=pass_id, resident=membership, residence=residence
+        )
+    except GuestPass.DoesNotExist:
+        raise ValidationError({"detail": "Pase no encontrado."})
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)